### PR TITLE
Add temperature dependant thermal expansion to tutorial

### PR DESCRIPTION
**Pull request by aeslaughter**

### CODE REVIEW

**Comment by aeslaughter** [comment]
@lindsayad If you have some time, can you take a look at step 9.

I updated everything that I can to AD, even tensor mechanics stuff. But, NEWTON doesn't seem to do all that great. Also, I switched to PJFNK with SMP/full=true and it fails completely.

File: tutorials/darcy_thermo_mech/doc/content/problem/step06.md
Comment (lindsayad) [inline tutorials/darcy_thermo_mech/doc/content/problem/step06.md]: Can be used to determine what?

File: tutorials/darcy_thermo_mech/step09_mechanics/problems/step9.i
Comment (lindsayad) [inline tutorials/darcy_thermo_mech/step09_mechanics/problems/step9.i]: So we saw that this made the solve significaintly better. For further debugging I would be curious how this does with `-pc_type lu`. For those time steps that we saw taking a lot of linear iterations, I would be curious to know how Jacobian-free is doing at that point. I'm wondering whether at that point whether our Jacobian-free approximation is doing a poor job of representing the Jacobian. It would be good to run with `Outputs/print_linear_residuals=false -ksp_monitor_true_residual` and see whether the true residual is dropping. Does this problem get more and more non-linear over time?

Comment (lindsayad) [reply tutorials/darcy_thermo_mech/step09_mechanics/problems/step9.i]: > Jacobian-free approximation is doing a poor job of representing the Jacobian

Sorry this would show up more in the non-linear iterations, not the linears. But you can see cases where the GMRES residuals drop but the true residuals do not; this usually indicates that your Jacobian-free process is non-linear, e.g. due to finite differencing of a noisy function or something. With good scaling though, that should be less of a concern.

Actually the most obvious explanation for a degrading linear solve would be if our off-diagonal terms are getting stronger, since we are currently only using the diagonals in the preconditioning matrix. Is that possibly happening? Won't happen in time for training, but we definitely want to figure out where our Jacobian bug is! Then we could run with NEWTON :smile:

Comment (aeslaughter) [reply tutorials/darcy_thermo_mech/step09_mechanics/problems/step9.i]: Thanks for digging into this, I think what I have in this PR is good for the training. We will get to this example sometime on Tue., so you might be able to help with this section and could talk about solver options and such.

**Comment by aeslaughter** [comment]
@permcody Can you take a quick look, this is phase 1 of the changes that I need to get in. I am working on the stuff from our meeting.

**Comment by aeslaughter** [comment]
@permcody There is a bit of a melt down with CIVET now and the stuff in next really needs to get in, so don't merge. If you approve I will merge when the storm ends.

**Review by permcody — APPROVED** [review]

File: tutorials/darcy_thermo_mech/step09_mechanics/problems/step9.i
Comment (aeslaughter) [reply tutorials/darcy_thermo_mech/step09_mechanics/problems/step9.i]: -pc_type lu works much better, it doesn't seem to get worse over time.

**Comment by lindsayad** [comment]
Hmm I wonder if the connective part just becomes too much for amg

**Pull request merged**
